analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/AnalysisScaffolding.cs

```diff
@@ -90,8 +91,9 @@ public static string CreateSonarProjectConfigWithFilesToAnalyze(TestContext cont
             string[] globalExclusions = null,
             string[] testExclusions = null,
             string[] testInclusions = null,
-            string[] globalTestExclusions = null) =>
-            TestHelper.WriteFile(context, "SonarLint.xml", GenerateSonarLintXmlContent(language, analyzeGeneratedCode, exclusions, inclusions, globalExclusions, testExclusions, testInclusions, globalTestExclusions));
+            string[] globalTestExclusions = null,
+            List<SonarLintXmlRule> rulesParameters = null) =>
```

Comment:
This is exploding too much. What about sending the `SonarLint` object here instead and serializing it? It is its very purpose.

Reply:
By "serializing" I mean 2 lines of code using serializer.

Reply:
Unfortunately, the SonarLintXml is unaware of the specific properties (analyzeGeneratedCode, exclusions, inclusions..etc). It knows only that Settings contains a list of Key/Value pairs. So to simplify this part we should pass every time the related keys (and not only the values) to the `GenerateSonarLintXmlContent`. Should we do that anyway?

Reply:
Or create a helper method that will do this one specific thing for you - to produce the object, that will be then serialized - in another PR.